frontend/src/reducer.test.ts

```diff
@@ -0,0 +1,136 @@
+
+  describe('messageReceived', () => {
+    it('adds new messages to existing town message chain', () => {
+      const messageToTest = createMessageForTesting(MessageType.TownMessage, nanoid());
+      const message2ToTest = createMessageForTesting(MessageType.TownMessage, nanoid());
+      const firstState = appStateReducer(createSampleAppState(), {
+        action: 'messageReceived',
+        message: messageToTest,
+      });
+      expect(firstState.townMessageChain.messages.length).toBe(1);
+      expect(firstState.townMessageChain.messages).toContain(messageToTest);
+      const secondState = appStateReducer(firstState, {
+        action: 'messageReceived',
+        message: message2ToTest,
+      });
+      expect(secondState.townMessageChain.messages.length).toBe(2);
+    });
+
+    it('adds new messages to existing proximity message chain', () => {
+      const messageToTest = createMessageForTesting(MessageType.ProximityMessage, nanoid());
+      const message2ToTest = createMessageForTesting(MessageType.ProximityMessage, nanoid());
+      const firstState = appStateReducer(createSampleAppState(), {
+        action: 'messageReceived',
+        message: messageToTest,
+      });
+      expect(firstState.proximityMessageChain.messages.length).toBe(1);
+      expect(firstState.proximityMessageChain.messages).toContain(messageToTest);
+      const secondState = appStateReducer(firstState, {
+        action: 'messageReceived',
+        message: message2ToTest,
+      });
+      expect(secondState.proximityMessageChain.messages.length).toBe(2);
+    });
+
+    it('creates new message chain for direct message', () => {
+      const messageToTest = createMessageForTesting(
+        MessageType.DirectMessage,
+        player1Id,
+        player2Id,
+      );
+      const firstState = appStateReducer(createSampleAppState(), {
+        action: 'messageReceived',
+        message: messageToTest,
+      });
+      const messageChainToCheck = firstState.directMessageChains[directMessageId];
+      expect(messageChainToCheck.messages.length).toBe(1);
+      expect(messageChainToCheck.messages).toContain(messageToTest);
+    });
+    it('adds message to existing chain for direct message', () => {
+      const messageToTest = createMessageForTesting(
+        MessageType.DirectMessage,
+        player1Id,
+        player2Id,
+      );
+      const message2ToTest = createMessageForTesting(
+        MessageType.DirectMessage,
+        player1Id,
+        player2Id,
+      );
+      const firstState = appStateReducer(createSampleAppState(), {
+        action: 'messageReceived',
+        message: messageToTest,
+      });
+      expect(firstState.directMessageChains[directMessageId].messages.length).toBe(1);
+      const secondState = appStateReducer(firstState, {
+        action: 'messageReceived',
+        message: message2ToTest,
+      });
+      expect(secondState.directMessageChains[directMessageId].messages.length).toBe(2);
+    });
```

Comment:
is there a test we could add for sending messages to the correct chain? If you had two ongoing direct message chains with two users

Reply:
just added one!